### ISSUE TITLE
support vehicle=* tag

### ISSUE THREAD

**Issue by matkoniecz** [openstreetmap/id-tagging-schema#378]
Recreation of https://github.com/openstreetmap/iD/issues/3945

In many cases `vehicle=` tagging is the best representation of situation on the ground

In particular: https://commons.wikimedia.org/wiki/File:Zakaz_ruchu_01.JPG sign is equivalent of `vehicle=no`, not `motor_vehicle=no`

See https://taginfo.openstreetmap.org/keys/vehicle https://wiki.openstreetmap.org/wiki/Key%3Avehicle

Ideally access field would allow setting this tag and if set it would also change display of `bicycle` and `motor_vehicle` unless this tags would be also set (like `access` is doing right now)

**Comment by 1ec5** [comment]
The logic of which access field affects which access field could get confusing to users who aren’t familiar with the tree on [the main access key page](https://wiki.openstreetmap.org/wiki/Key:access#Transport_mode_restrictions). The discussion in openstreetmap/iD#3945 went in the direction of a nested tree of fields (simplified from the wiki page). Alternatively, iD could present a StreetComplete-like palette of access restriction signs to choose from. However, either approach would require changes to the field implementation in the editor itself, not just a simple change to the field definition in id-tagging-schema.

**Comment by matkoniecz** [comment]
> The logic of which access field affects which access field could get confusing to users who aren’t familiar with the tree on the main access key page. 

Are you sure? Is the worry that users would be confused by bicycle treated as a vehicle and affected by limiting vehicles?

**Comment by 1ec5** [comment]
Yes, because words like “vehicle” have different meanings to different people. Many laypeople consider “vehicle” to mean “motor vehicle”. If “Vehicles” appears along with “Motor Vehicles”, the user might intuit that “Vehicles” excludes bicycles, but [does it include horses](https://www.quora.com/Is-a-horse-a-vehicle-or-are-vehicles-only-things-we-build-to-ride)?

I think it would be helpful to make the connections visually clear. Currently, the only row that affects the other rows is the general one for All, which is self-explanatory. Short of indentation, rearranging the rows to be strictly general to specific might help.

**Comment by matkoniecz** [comment]
Good point about Vienna Convention vs MUTCD.

**Comment by tyrasd** [comment]
As a very first step towards adding support for this, I've modified iD's `access` field(s) to at least display the implied/derived values of the `motor_vehicle` and `bicycle` fields if a value for `vehicle` is already mapped. For example, a street with `access=permissive + vehicle=no` would be now shown as follows:

<img src="https://user-images.githubusercontent.com/1927298/152407368-2616a20c-a95b-48c8-8df9-f6f60fd41115.png" width="300px">

This somewhat improves the situation as existing features are now less wrongly represented in the table of the access field. But it is of course also not a final solution for the issue.

**Comment by Hufkratzer** [comment]
>  [does “Vehicles” include horses](https://www.quora.com/Is-a-horse-a-vehicle-or-are-vehicles-only-things-we-build-to-ride)?

No.  From a legal point of view a horse is not a vehicle, at least not in Germany.

**Comment by tas50** [comment]
A horse is legally a vehicle in the US.

**Comment by simonpoole** [comment]
What is the problem with simply providing vehicle support for Vienna Convention countries, and let the US sabotage itself. It seems weird that one of the, if not most common access restriction cannot be entered outside of the raw tag editor, and that because of a country that doesn't need it.

**Comment by 1ec5** [comment]
> To be clear, I don’t mean to argue against the inclusion of a “Vehicles” row. I just think it would be helpful to make the relationships between these keys visually clear, for the same reason that the wiki presents the keys in a tree. Currently, the only row that affects the other rows is the general one for All, which is self-explanatory. Short of indentation, rearranging the rows to be strictly general to specific might help.

Rearranging the rows wouldn’t be bad for Vienna Convention countries, would it?

**Comment by simonpoole** [comment]
> I just think it would be helpful to make the relationships between these keys visually clear, for the same reason that the wiki presents the keys in a tree.

I didn't comment on that, because, dirty secret, IRL access for vehicle classes are as a rule not defined hierarchically and even when they are, the positions can change depending on context/country/etc. So any order is likely as good as any other.

**Comment by 1ec5** [comment]
Thanks, I didn’t know that! In that case, should the logic in openstreetmap/iD@f7d1459c97a249c81aea07f292d56cf98be7cb2d depend on the country or context too?

**Comment by simonpoole** [comment]
IMHO not worth the trouble, for example for most countries the Vienna convention sign/rule for no motor vehicle access includes mopeds, but in some it doesn't. This is partially solvable with region specific presets, but  in any case the assumption that "vehicle" = motor_vehicle + bicycle would seem to be wrong, as in not including everything that is considered a vehicle.

**Comment by tordans** [comment]
@tyrasd would you merge a PR that adds a regional preset as described above? I am very in favor of having a simple fix now before a bigger project is started; that can still happen regardless. Working on the traffic_sign <> osm tagging project I run into this issue a lot.
